src/auth/digest/Config.cc

```diff
@@ -826,11 +826,14 @@ Auth::Digest::Config::decode(char const *proxy_auth, const HttpRequest *request,
             break;
 
         case DIGEST_NC:
-            if (value.size() != 8) {
+            if (value.size() == 8) {
+                // For historical reasons, the nc value MUST be exactly 8 hexadecimal
+                // digits.
+                xstrncpy(digest_request->nc, value.rawBuf(), value.size() + 1);
+                debugs(29, 9, "Found noncecount '" << digest_request->nc << "'");
+            } else {
                 debugs(29, 9, "Invalid nc '" << value << "' in '" << temp << "'");
```

Comment:
AFAICT, this code is inside a loop, and, hence, might see a dozen different `nc` values, some valid, some not. The last value wins. Unless it is invalid (then some earlier value wins; if any). Such inconsistencies often trigger bugs. Unfortunately, this code only validates values _after_ they are stored, but we can at least ensure that belated validation code always looks at the last seen value (or nothing):

```suggestion
                debugs(29, 9, "Invalid nc '" << value << "' in '" << temp << "'");
                digest_request->nc[0] = 0;
```